dockerfiles/Dockerfile.demo.fips

```diff
@@ -0,0 +1,20 @@
+ARG GO_VERSION
```

Comment:
I would need to play with it a bit to see if this would work, but I'd prefer if we could add this logic to the existing Dockerfile instead of keeping this separate. The only hint I can offer is the end of this Dockerfile where we pick the base image for the right OS, but there might be a better way.

Reply:
I think the big blocker is going to be the fact that the mariner images aren't multi-platform AND there are different package install commands necessary (e.g. `yum install -y make ca-certificates`). I'm not a Docker build expert by any means, but unless there's conditional logic in Dockerfile syntax that allows switching between final images (e.g. `mcr.microsoft.com/cbl-mariner/distroless/base:2.0` and `gcr.io/distroless/static`) as well as selecting only a single builder image based on args, then these may have to stay separate